woocommerce/woocommerce-gateway-stripe
# Hide refund button for non-captured orders

Currently, when merchants have the "capture later" setting enabled, a refund button is displayed for new orders which are authorized but not captured. However, this makes no sense because there's no such thing as refunding a charge that has not been made yet.

The current behavior is that if they do ANY refund on an uncaptured order, full or partial, it will simply cancel the order and release the pre-authorization, but the UI does a poor job of indicating that and the behavior is inherently unintuitive.

This change simply checks if the payment for the order has been authorized but not charged, and if so, hides the refund button and replaces it with a disabled "Refunding unavailable" button which includes a tooltip which explains why.

Fixes #2338.

## Testing instructions

- Check out branch
- Go to WooCommerce Settings -> Payments -> Stripe -> Settings
- Under Transaction preferences, enable `Issue an authorization on checkout, and capture later`, click save changes
- Place an order
- In the admin, go to WooCommerce -> Orders, click the order you just created
- In the second section that shows the items in the order, you should **not** see the "Refund" button, but instead see a disabled "Refunding unavailable" button with a tooltip that displays additional info:
![](https://caps.evan.pro/a17503.png)

Finally, disable the `Issue an authorization on checkout, and capture later`, place another order, and make sure the normal refund button is still displayed and functions properly.
